src/node_http2.cc

```diff
@@ -879,6 +887,17 @@ ssize_t Http2Session::ConsumeHTTP2Data() {
   // We are done processing the current input chunk.
   DecrementCurrentSessionMemory(stream_buf_.len);
   stream_buf_offset_ = 0;
+  if (!stream_buf_ab_.IsEmpty()) {
+    ArrayBuffer* array_buffer_to_detach =
+        stream_buf_ab_.template ClearWeak<ArrayBuffer>();
+    if (array_buffer_to_detach != nullptr) {
+      // TODO(thangktran): drop this check when V8 is pumped to 8.0 .
+      if (!array_buffer_to_detach->IsExternal())
+        array_buffer_to_detach
+            ->Externalize(array_buffer_to_detach->GetBackingStore());
+      array_buffer_to_detach->Detach();
+    }
+  }
```

Comment:
Uh … I think detaching the `stream_buf_ab_` would invalidate stream chunks emitted to JS? Can you explain these changes here?

Reply:
sure. so `stream_buf_ab_` is the only `ArrayBuffer` from `Http2Session` so i believe the error has to do with it. So i want to make sure that the `ArrayBuffer` under `stream_buf_ab_` is detached right before `Reset()` is called.

Reply:
> so `stream_buf_ab_` is the only `ArrayBuffer` from `Http2Session`/`node_http2.cc` so i believe the error has to do with it.

There’s the one that’s being changed here above, too (the one for `js_fields_`). I think that’s the problematic one, because we allocate the `Http2Session` ourselves but don’t invalidate/detach the `ArrayBuffer` when the underlying `Http2Session` memory is freed.

I think that means that it would need to be stored on the `Http2Session` in a `v8::Global`, yes.

> I ran all tests and they all passed, did i miss something?

Hm, that’s surprising … maybe our tests just don’t verify this :/ I’ll see if I can come up with something for that…

Reply:
> > I ran all tests and they all passed, did i miss something?
> 
> Hm, that’s surprising … maybe our tests just don’t verify this :/ I’ll see if I can come up with something for that…

I did some digging here and it turns out this was still wroking because `stream_buf_ab_` is never weak, so `ClearWeak()` always returned `nullptr`, and the array buffer never actually got detached.

Reply:
That means `js_fields_` is our main suspect here? I'm working on a fix for this.

Reply:
Yes, exactly – I haven’t verified that it’s *the* thing that caused the crashes earlier in CI, but it is something that *could* cause crashes.

Reply:
fixed.

Reply:
Can you remove the block of changes here? As mentioned above, it doesn’t do anything right now :)